### PR TITLE
major mosart refactor

### PR DESCRIPTION
**Pull request by mvertens**
Removed all references to rtm
- files have been renamed and namelists no longer contain rtm in the name

New modularity: 
- introduced new modules with new derived types and methods
  - mosart_control_type.F90
  - mosart_tctl_type.F90
  - mosart_tparameter_type.F90
  - mosart_tspatialunit_type.F90
  - mosart_tstatusflux_type.F90
- the new modules modularize a lot of the complexity and variables that were previously found in RunOffMod.F90 and permit decomposition initialization to be more flexible and transparent.

New halo capability
- Ability to have halo regions and communication using ESMF. This is needed for computing derivatives in upcoming new additions to MOSART.
- New halo namelist - use_halo_option. When this is set to true halos can be activated. See the `test_halo` subroutine in `mosart_control_type.F90` module.
- Verified that the results for the halos are bfb identical regardless of the number of processors that are used.

Testing:

Used the following sandbox:
https://github.com/mvertens/NorESM/pull/new/feature/noresm2_5_alpha01_mosart

Ran the following test and confirmed that baselines were bfb with mosart1_0_49_noresm_v1.
 ./create_test --xml-category mosart_noresm --xml-machine betzy --generate mosart1_0_49_noresm_v2 --compare mosart1_0_49_noresm_v1 --baseline-root ~/baselines --project nn9039k

### CODE REVIEW

File: src/riverroute/mosart_data.F90
Comment (mvdebolskiy) [inline src/riverroute/mosart_data.F90:1]: Files like this have a bit of ambiguous names. It's not quite clear which ones just contain types and which ones are modules with subroutines. Also, there are comments missing on the contents of the module right after the module name. Do you want me to give collab rights on that branch, so I try to fix some of this?

**Comment by mvertens** [comment]
@mvdebolskiy - could you please review this or simply just okay it given that we reviewed the PR to follow this for liquid tracers. I'd like for this PR to come into the externals for noresm2_5_alpha02 which I am putting together.

**Review by mvdebolskiy — APPROVED** [review]

**Comment by mvertens** [comment]
@mvdebolskiy - You should have collaborator rights to my fork so you should be able to
push back. Thanks for offering to do this! Not sure why I have conflicts -
I didn't see this until now.

**Comment by mvertens** [comment]
@mvdebolskiy - did you rerun the test suite - or should I in order to merge this PR?

**Comment by mvertens** [comment]
@mvdebolskiy - I just reran the test suite with your changes and everything is bfb. So I'm going to merge this PR.

**Pull request merged**